features/dd-sdk-android-rum/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumResourceScope.kt

```diff
@@ -65,6 +67,12 @@ internal class RumResourceScope(
     private var statusCode: Long? = null
     private var size: Long? = null
 
+    init {
+        networkSettledMetricResolver.resourceWasStarted(
+            NetworkSettledResourceContext(resourceId, eventTime.nanoTime)
+        )
+    }
```

Comment:
> [!NOTE]
> Why are we splitting the computation in both the `RumViewScope` and `RumResourceScope`. All Resource related event do go through the `RumViewScope` first (see the `RumViewScope.onStartResource()` method), so it might be simpler to compute the network settled directly via the `RumViewScope`. It also might be simpler to compute it in the `RumViewScope` instead of a `NetworkSettledMetricResolver`, wdyt?